index.js

```diff
@@ -1,19 +1,42 @@
 const searchController = require('./app/controllers/searchController');
+const airportDetailsController = require('./app/controllers/airportDetailsController');
+
 exports.handler = async (event) => {
     // if there is nothing in querystring, then just pass an empty obj instead of undefined
     const filter = (event && event.params && event.params.querystring) ? event.params.querystring : {};
-    console.log('below is the filter');
-    console.log(filter);
-
-  const promise = new Promise(function(resolve, reject) {
-        searchController.get(filter, (err,result) => {
-            if(err) {
-                console.log(err);
-                reject(Error(err));
-            } else {
-                resolve(result);
-            }
+    const path = event.context["resource-path"];
+    if (path === '/search') {
+        return new Promise(function(resolve, reject) {
```

Comment:
This boilerplate is even clunkier when repeated four times in a row than for a single occurrence, and I can see some obvious candidates for refactoring to reduce code duplication.

Reply:
Yes, however, with the short about of time we had. I just wanted to make sure that it would work first.

Reply:
Using a `switch` statement helps a *very* little, but you're still repeating the same eight or ten lines for every single route, only differing in one of them as to what method to call to handle the request.

Reply:
Yes, you are right.